transaction/src/manifest/parser.rs

```diff
@@ -364,6 +370,8 @@ impl SborValueKindIdent {
 pub struct Parser {
     tokens: Vec<Token>,
     current: usize,
+    max_depth: usize,
```

Comment:
@dhedey you suggested to add the depth tracking into the `generator`. But I've found out that the `parser` might also cause stack overlow issues.
Therefore I've added tracking here.
Decided to not track the depth in `generator` additionally since they're both used only in compiler. But if you want to add these checks in generator, I will be more than happy to do it :)